### PR TITLE
Exception_static_build defalt: Partial revert of dabbeada3a

### PR DESCRIPTION
**Pull request by carlopi**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -276,8 +276,11 @@ option(BUILD_MAIN_DUCKDB_LIBRARY
         TRUE)
 option(EXTENSION_STATIC_BUILD
         "Extension build linking statically with DuckDB. Required for building linux loadable extensions."
-        TRUE)
+        FALSE)
 
+if(WIN32)
+  set(EXTENSION_STATIC_BUILD TRUE)
```
Comment (samansmink) [inline CMakeLists.txt:282]: Why is this turned on by windows by default?

**Review by samansmink — COMMENTED** [review]
@carlopi thanks for finding this. I found that this is most likely caused by the `test/extension` tests. With the EXTENSION_STATIC_BUILD flag these extensions get very large (~260MB). I'm not exactly sure how this fails on CI because locally it seems to run fine. It could be a OOM thing, but the github MacOS runners are relatively beefy with 14GB memory so not sure.  

I think reverting back to default off for this option is probably the best solution here, @Mytherin?

File: CMakeLists.txt (hunk 1 above)
Comment (Mytherin) [reply CMakeLists.txt:282]: We would have to revert the entire PR (including adding back support for delayload) to support disabling static builds on Windows again

**Comment by Mytherin** [comment]
Thanks! Indeed we need to investigate this further later - the size on debug mode is excessive

**Pull request merged**